azurerm/resource_arm_mysql_server.go

```diff
@@ -135,7 +135,7 @@ func resourceArmMySqlServer() *schema.Resource {
 							Type:         schema.TypeInt,
 							Required:     true,
 							ForceNew:     true,
-							ValidateFunc: validateIntBetweenDivisibleBy(5120, 1048576, 1024),
+							ValidateFunc: validateIntBetweenDivisibleBy(5120, 4194304, 1024),
```

Comment:
Shouldn't this also validate the SKU since only general purpose and memory optimized support this larger upper limit?

Reply:
I was in favor of removing the limit check altogether. The server anyway fails and gives a proper error message when we limit sizes.

Reply:
Good catch, i agree

Reply:
@jeffreyCline I'm not super familiar with your conventions / golang style best practices. would you use a case/switch or an if here? Or is there a better option? Is there an example of another resource in the provider that does this?

Reply:
@gabinante,

[Here](https://github.com/terraform-providers/terraform-provider-azurerm/blob/4518e2bf0a02fcc7f4dd4e66e5fe7440538cb841/azurerm/resource_arm_automation_schedule.go#L113) is an example of using the a `CustomizeDiff` to ensure 1 of 2 properties is set.

In your case you can use this to get the `storage_mb` property is within the bounds of what the current `sku` allows.

Reply:
@gabinante I went ahead and added the validation code directly to your branch, hope that was ok. 😃